pixeltable/functions/reve.py

```diff
@@ -0,0 +1,259 @@
+                        raise pxt.Error(
+                            f'Reve request {request_id} expected content type {requested_content_type}, '
+                            f'got {resp.content_type}'
+                        )
+
+                    img_data = await resp.read()
+                    if len(img_data) == 0:
+                        raise pxt.Error(f'Reve request {request_id} resulted in an empty image')
+                    img = PIL.Image.open(BytesIO(img_data))
+                    _logger.debug(
+                        f'Reve request {request_id} successful. Image bytes: {len(img_data)}, size: {img.size}'
+                        f', format: {img.format}, mode: {img.mode}'
+                    )
+                    return img
+                case 429:
+                    # Try to honor the server-provided Retry-After value if present
+                    # Note: Retry-After value can also be given in the form of HTTP Date, which we don't currently
+                    # handle
+                    retry_after_seconds = None
+                    retry_after_header = resp.headers.get('Retry-After')
+                    if retry_after_header is not None and re.fullmatch(r'\d{1,2}', retry_after_header):
+                        retry_after_seconds = int(retry_after_header)
+                    _logger.info(
+                        f'Reve request {request_id} failed due to rate limiting, retry after header value: '
+                        f'{retry_after_header}'
+                    )
+                    # This error message is formatted specifically so that RequestRateScheduler can extract the retry
+                    # delay from it
+                    raise pxt.Error(
+                        f'Reve request {request_id} failed due to rate limiting (429). retry-after:'
+                        f'{retry_after_seconds}'
+                    )
+                case _:
+                    _logger.error(
+                        f'Reve request {request_id} failed with status code {resp.status} and error code {error_code}'
+                    )
+                    raise pxt.Error(f'Reve request failed with status code {resp.status} and error code {error_code}')
+
+
+@register_client('reve')
+def _(api_key: str) -> _ReveClient:
+    return _ReveClient(api_key=api_key)
+
+
+def _client() -> _ReveClient:
+    return Env.get().get_client('reve')
+
+
+# TODO Regarding rate limiting: Reve appears to be going for a credits per minute rate limiting model, but does not
+# currently communicate rate limit information in responses. Therefore neither of the currently implemented limiting
+# strategies is a perfect match, but "request-rate" is the closest. Reve does not currently enforce the rate limits,
+# but when it does, we can revisit this choice.
+@pxt.udf(resource_pool='request-rate:reve')
+async def create(
+    prompt: str, *, format: str, aspect_ratio: str | None = None, version: str | None = None
```

Comment:
Do we really want user-specified image formats as UDF parameters? None of our other UDFs have that, and since it's being returned initially as an in-memory PIL image it's not clear what it means. There's an argument that if we do want user-specified image formats, they should be controlled at the column definition level rather than as UDF parameters. I think this merits some discussion.

cc @mkornacker

Reply:
offline discussion: 
1. drop format as a UDF parameter, always use png for now
2. save to disk using the default format, which is webp or jpeg at the moment.